### PR TITLE
improve url regex

### PR DESCRIPTION
**Pull request by mamico**
Improved the url regex to allow non-public urls, such as https://intranet/ or file://server/share, e.g., for intranet.

### CODE REVIEW

**Comment by davisagli** [comment]
@mamico Makes sense. Needs a towncrier entry before it's ready to merge.

File: packages/volto/news/6186.feature
Comment (stevepiercy) [inline packages/volto/news/6186.feature:1]: We need to wrap URLs with backticks to make these inline literals, else linkcheckbroken attempts to visit the URL and fails.

```suggestion
Improved URL regex to allow non-public or intranet URLs, such as `https://intranet/` or `file://server/share`. @mamico
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Update news.

**Review by stevepiercy — APPROVED** [review]
@davisagli already approved in a comment in this PR, so I'm approving, too, so that it can be merged when CI is all green.

**Pull request merged**
